mack/__init__.py

```diff
@@ -735,3 +736,119 @@ def rename_delta_table(
         delta_table.toDF().write.format("delta").mode("overwrite").saveAsTable(
             new_table_name
         )
+
+def type_3_scd_upsert(
```

Comment:
How would the user use this function with an `effective_date` column?

Reply:
I recomend to use it as optional parameter.

If needed, the function have to update the `effective_date` each time a real update occurs on the current and previous column.

A structure of a delta table with an `effective_date` could be something like this: 

+----+----+----+-------+------------------+
|pkey|name|job|prev_job|job_effective_date
+----+----+---+--------+------------------+

or much more wider 
 
+----+----+----+-------+------------------+-------+------------+-----------------+---------+--------------+------------------------+
|pkey|name|job|prev_job|job_effective_date|country|prev_country|country_effective|continent|prev_continent|continent_effective_date
+----+----+---+--------+------------------+-------+------------+-----------------+---------+--------------+------------------------+

Now how we will provide the column names representing the `effective_date` to the function ? 

A parameter as a list of list ? --> param = ((`job`,`prev_job`,`job_effective_date`),(`country`,`prev_country`,`country_effective`),(`continent`,`prev_continent`,`continent_effective_date`))

A 3 parameters: `current_col_names`, `previous_col_names` and `effective_date_col_names` ? 

I don't like the second option unless we consider that the function cannot handle multiple scd columns at a time.